lib/Frontend/Frontend.cpp

```diff
@@ -478,8 +478,49 @@ void CompilerInstance::setupOutputBackend() {
         *CAS, *ResultCache, *CompileJobBaseKey,
         Invocation.getFrontendOptions().InputsAndOutputs,
         Invocation.getFrontendOptions().RequestedAction);
+
+    struct FilteringOutputBackend : public llvm::vfs::ProxyOutputBackend {
```

Comment:
Why do you re-implement this? `llvm::vfs::makeFilteringOutputBackend` is available through VirtualOutputBackends.h